### PR TITLE
[SPARK-17162] Range does not support SQL generation

### PR DESCRIPTION
**Pull request by ericl**
## What changes were proposed in this pull request?

The range operator previously didn't support SQL generation, which made it not possible to use in views.
## How was this patch tested?

Unit tests.

cc @hvanhovell

### CODE REVIEW

**Comment by rxin** [comment]
Can you make the options for logical plan optional?

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala [hunk 1]
```diff
@@ -205,6 +205,9 @@ class SQLBuilder private (
     case p: ScriptTransformation =>
       scriptTransformationToSQL(p)
 
+    case Range(start, end, step, numPartitions, output) =>
+      s"SELECT id AS `${output.head.name}` FROM range($start, $end, $step, $numPartitions)"
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala]: while you are at it, can you move this into a toSQL function in Range?

**Comment by ericl** [comment]
Updated, ptal

**Comment by hvanhovell** [comment]
LGTM - pending jenkins.

**Comment by rxin** [comment]
Merging in master/2.0.

**Pull request closed without merging**